privval/signer_listener_endpoint.go

```diff
@@ -0,0 +1,283 @@
+package privval
+
+import (
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	cmn "github.com/tendermint/tendermint/libs/common"
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+// SignerValidatorEndpointOption sets an optional parameter on the SocketVal.
+type SignerValidatorEndpointOption func(*SignerListenerEndpoint)
+
+// SignerListenerEndpoint listens for an external process to dial in
+// and keeps the connection alive by dropping and reconnecting
+type SignerListenerEndpoint struct {
+	cmn.BaseService
+
+	mtx      sync.Mutex
+	listener net.Listener
+	conn     net.Conn
+
+	timeoutAccept    time.Duration
+	timeoutReadWrite time.Duration
+
+	stopCh, stoppedCh chan struct{}
```

Comment:
cmn.BaseService exposes `Quit()` method returning a channel, which is closed when the service is stopped